repository/utils.go

```diff
@@ -0,0 +1,77 @@
+package repository
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/go-github/v36/github"
+)
+
+// removeDuplicate removes duplicate repositories from the input list and returns a new slice of unique repositories.
+// It checks for duplicates based on the combination of the "Owner" and "Name" attributes
+func removeDuplicate(inputList []Repository) []Repository {
```

Comment:
you might want to consider using the "new" https://pkg.go.dev/slices#DeleteFunc func (note that it requires go 1.21, so you'll need to update the go.mod file too. which is a good thing, because go 1.19 is not supported anymore)

Reply:
It seems that DeleteFunc expects a slice with a type "E" that satisfies the "comparable" constraint. On our case, Repository has a map field that can't be used directly because not considered as "comparable" in go

Reply:
done